### ISSUE TITLE
Provide Summary after tflint Command Run

### ISSUE THREAD

**Issue by fracca** [terraform-linters/tflint#1849]
When running the **tflint** command within a pipeline, I want a summary report on successful and failed runs.

The Summary report should include

* Count of Rules checked(for each enabled plugin)

If the report contains failures:
* Count of each error type (Errors: nnn, Warnings: nnn, etc...)

The reason for this is to confirm that all rules have been run. This provides confidence in the Linting.

**Comment by fracca** [comment]
Current JUNIT outputs show this only - (so may be a bug)
`<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite tests="0" failures="0" time="0" name="">
    <properties></properties>
  </testsuite>
</testsuites>`

**Comment by wata727** [comment]
Is this about JUnit formatter?

If so, this is the intended design. Rules (linter) and tests are different concepts and cannot be directly mapped to JUnit output format. For example, one rule may detect multiple issues. In that case, the sum of `failures` may be much larger than `tests`.

See also https://github.com/terraform-linters/tflint/discussions/1600 https://github.com/terraform-linters/tflint/issues/1354

On the other hand, if you're talking about the default formatter, I think this request makes sense.

See also https://github.com/terraform-linters/tflint/discussions/1758

**Comment by fracca** [comment]
Yes, the original request around the default formatter is why I started this. 

I was playing with the JUnit formatter to see if this would be acceptable to generate an appropriate test output for BitBucket Pipelines to latch onto.

**Comment by wata727** [comment]
Thank you for your clarification. Perhaps this can be achieved by changing the `formatter/pretty.go`.
https://github.com/terraform-linters/tflint/blob/1d58c87ecb3ca18f13ac2e3c96cdad5be3e02354/formatter/pretty.go#L24-L30

The count of rules is not passed to the formatter, so we'll have to fix it more upstream, but it shouldn't be too difficult.

**Comment by wata727** [comment]
Before getting into this, we should keep in mind that some people may want to empty the output if no issues. See https://github.com/terraform-linters/tflint/issues/182
There is room to consider whether to add the `--quiet` option again or enable it with an option.

Also, there is currently no way to get a list of enabled rules, so we will need to extend the SDK to accomplish this.

**Comment by PatMyron** [comment]
hacky stopgap before we get around to proper support:
https://github.com/aws-cloudformation/cfn-lint/issues/882#issuecomment-590121795
```shell
tflint --format=sarif | grep -Eo '"ruleId.*' | sort | uniq -c | sort -nr
      9 "ruleId": "terraform_required_providers",
      3 "ruleId": "terraform_unused_declarations",
```
